lib/Parse/ParseDecl.cpp

```diff
@@ -3285,7 +3285,7 @@ ParserResult<PoundDiagnosticDecl> Parser::parseDeclPoundDiagnostic() {
     // Catch #warning(oops, forgot the quotes)
     SourceLoc wordsStartLoc = Tok.getLoc();
 
-    while (!Tok.isAtStartOfLine() && Tok.isNot(tok::r_paren)) {
+    while (!Tok.isAtStartOfLine() && !Tok.isAny(tok::r_paren, tok::eof)) {
```

Comment:
`Tok.isNot(tok::r_paren, tok::eof)` should also work, but I personally prefer `!Tok.isAny(` 👍